### PR TITLE
Extend ProductType with type

### PR DESCRIPTION
**Pull request by IKarbowiak**
- Extend `ProductType` model with `kind` field
- Extend `ProductType` type with `kind` field
- Extend `ProductTypeInput` input with `kind` field

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3879 Extend ProductType](https://app.clickup.com/t/2549495/SALEOR-3879)

**Review by tomaszszymanski129 — APPROVED** [review]
Codewise looks fine, but i would reconsider the naming `ProductTypeType`, sounds strange and seems confusing to me.
Maybe something like `ProductTypeKind` / `ProductTypeFamily` / `ProductTypeClass` ?
Perhaps someone else could advise here :)

**Comment by IKarbowiak** [comment]
@tomaszszymanski129 I agree, `ProductTypeKind` sounds good for me.

**Review by tomaszszymanski129 — APPROVED** [review]

File: saleor/product/models.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def __repr__(self) -> str:
 class ProductType(ModelWithMetadata):
     name = models.CharField(max_length=250)
     slug = models.SlugField(max_length=255, unique=True, allow_unicode=True)
+    type = models.CharField(max_length=32, choices=ProductTypeKind.CHOICES)
```
Comment (maarcingebala) [inline saleor/product/models.py:110]: Shouldn't we name this field `kind` for consistency?

**Review by korycins — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**